src/utils.jl

```diff
@@ -458,8 +458,8 @@ function flatten1(ex)
   for x in ex.args
     isexpr(x, :block) ? append!(ex′.args, x.args) : push!(ex′.args, x)
   end
-  # Don't use `unblock` to preserve line nos
-  return length(ex′.args) == 1 ? ex′.args[1] : ex′
+  # Don't use `unblock` to preserve line nos. Check for blocks that contain only a line number node.
+  return length(ex′.args) == 1 && !isline(ex′.args[1]) ? ex′.args[1] : ex′
```

Comment:
I'd say that in general, `Expr(:block, x)` -> `x` is a valid transform, including for line numbers, so I'm a bit reluctant.

Reply:
Are you saying that you prefer that try is special cased here?

Reply:
Could you special-case `try`? I'm not super into this code, but it looks to me like the special-casing solution would involve replacing the `flatten` `postwalk` with an explicit tree traversal, which I also wouldn't like very much. I don't know! It's not a fun bug to fix. Let's see if it really fixes https://github.com/JuliaLang/julia/issues/50710#issuecomment-1657190518. If not, then is there a reason to merge this? It might break someone else's subtle macrology.